### ISSUE TITLE
Refactoring: use pathlib

### ISSUE THREAD

**Issue by megies** [obspy/obspy#2751]
For the long term, we could clean up the code base a bit by internally using pathlib instead of `os.path.join(....)` and consorts. This is no vital or overly important task but it will help new contributors, to use it in new contributions and not introduce more old-style code, looking at the rest of the code base.

This is a pretty trivial task and is a good first issue for new people.

See e.g. https://github.com/obspy/obspy/pull/2740#discussion_r525934680

**Comment by dbgraybeal** [comment]
Do you know whether anyone is currently working on this? I've wanted to contribute Pathlib support to Obspy for a while now, and only just came across this Issue.

**Comment by heavelock** [comment]
Recently there was a PR on adding support for pathlib to read methods: https://github.com/obspy/obspy/pull/2743
However, the core of that issue was not touched there so feel free to give it a shot.

**Comment by dbgraybeal** [comment]
Ah, thank you for linking to that! I've apparently been assuming that pathlib Paths still needed to be converted without actually checking. 

Still, if this would be a useful, though not urgent, update to the code base, It would be nice to start contributing.

**Comment by heavelock** [comment]
Yeah, definitely it would be very useful! So feel free to submit some PRs!

**Comment by fAndreuzzi** [comment]
Maybe this could be closed? #2833
